react/src/components/UserProfileSettingModal.tsx

```diff
@@ -22,212 +48,140 @@ const UserProfileSettingModal: React.FC = () => {
+  const [userInfo, userMutations] = useCurrentUserInfo();
+  const [fetchKey, updateFetchKey] = useUpdatableState('initial-fetch');
+  const deferredMergedFetchKey = useDeferredValue(fetchKey);
+
+  const { user } = useLazyLoadQuery<UserProfileSettingModalQuery>(
+    graphql`
+      query UserProfileSettingModalQuery(
+        $email: String!
+        $isNotSupportTotp: Boolean!
+      ) {
+        user(email: $email) {
+          id
+          totp_activated @skipOnClient(if: $isNotSupportTotp)
+          ...TOTPActivateModalFragment
+        }
+      }
+    `,
+    {
+      email: userInfo.email,
+      isNotSupportTotp: !totpSupported,
     },
     {
-      suspense: false,
+      fetchPolicy: 'network-only',
+      fetchKey: deferredMergedFetchKey,
     },
   );
-  const totpActivated = userInfo?.user.totp_activated;
 
-  const { data: keyPairInfo } = useTanQuery(
-    'keyPairInfo',
+  const { data: keyPairs } = useTanQuery<
+    {
+      secret_key: string;
+      access_key: string;
+    }[]
+  >(
+    'baiClient.keypair.list',
     () => {
-      return baiClient.keypair.list(
-        baiClient.email,
-        ['access_key', 'secret_key'],
-        true,
-      );
+      return baiClient.keypair
+        .list(userInfo.email, ['access_key', 'secret_key'], true)
+        .then((res: any) => res.keypairs);
     },
     {
-      suspense: false,
+      suspense: true,
     },
   );
-  const selectOptions: SelectProps['options'] = [];
-  if (keyPairInfo) {
-    for (let i = 0; i < keyPairInfo.keypairs.length; i++) {
-      selectOptions.push({
-        value: keyPairInfo.keypairs[i].secret_key,
-        label: keyPairInfo.keypairs[i].access_key,
-      });
-    }
-    const matchLoggedAccount = _.find(keyPairInfo.keypairs, [
-      'access_key',
-      baiClient._config.accessKey,
-    ]);
-    form.setFieldsValue({
-      access_key: matchLoggedAccount?.access_key,
-      secret_key: matchLoggedAccount?.secret_key,
-    });
-  }
-
-  const { value, dispatchEvent } = useWebComponentInfo();
-  let parsedValue: {
-    isOpenUserPrefDialog: boolean;
-  };
-  try {
-    parsedValue = JSON.parse(value || '');
-  } catch (error) {
-    parsedValue = {
-      isOpenUserPrefDialog: false,
-    };
-  }
-
-  const { isOpenUserPrefDialog } = parsedValue;
-
-  const mutationToUpdateUserFullName = useTanMutation({
-    mutationFn: (values: { email: string; full_name: string }) => {
-      return baiClient.update_full_name(values.email, values.full_name);
-    },
-  });
 
-  const mutationToUpdateUserPassword = useTanMutation({
-    mutationFn: (values: {
-      old_password: string;
-      new_password: string;
-      new_password2: string;
-    }) => {
-      return baiClient.update_password(
-        values.old_password,
-        values.new_password,
-        values.new_password2,
-      );
+  const mutationToRemoveTotp = useTanMutation({
+    mutationFn: (email: string) => {
+      return baiClient.remove_totp(email);
```

Comment:
```suggestion
    mutationFn: () => {
      return baiClient.remove_totp();
```

It is not possible for general users to remove totp using email.